libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -510,3 +630,246 @@ void ContractLevelChecker::checkBaseABICompatibility(ContractDefinition const& _
+	}
+
+	return resolved;
+}
+
+
+void ContractLevelChecker::checkModifierOverrides(FunctionSet const& _funcSet, ModifierSet const& _modSet, std::vector<ModifierDefinition const*> _modifiers)
+{
+	for (ModifierDefinition const* modifier: _modifiers)
+	{
+		if (contains_if(_funcSet, MatchByName{modifier->name()}))
+			m_errorReporter.typeError(
+				modifier->location(),
+				"Override changes function to modifier."
+			);
+
+		auto [begin,end] = _modSet.equal_range(modifier);
+
+		// Skip if no modifiers found in bases
+		if (begin == end)
+			continue;
+
+		if (!modifier->overrides())
+			overrideError(*modifier, **begin, "Overriding modifier is missing 'override' specifier.");
+
+		for (; begin != end; begin++)
+			if (ModifierType(**begin) != ModifierType(*modifier))
+				m_errorReporter.typeError(
+					modifier->location(),
+					"Override changes modifier signature."
+			);
+	}
+
+}
+
+void ContractLevelChecker::checkOverrideList(FunctionSet const& _funcSet, FunctionDefinition const& _function)
+{
+	set<ContractDefinition const*> specifiedContracts =
+		_function.overrides() ?
+		resolveOverrideList(*_function.overrides()) :
+		set<ContractDefinition const*>{};
+
+	// Check for duplicates in override list
+	if (_function.overrides() && specifiedContracts.size() != _function.overrides()->overrides().size())
+	{
+		// Sort by name to find duplicate for error reporting
+		vector<ASTPointer<UserDefinedTypeName>> list =
+			sortByName(_function.overrides()->overrides());
+
+		// Find duplicates and output error
+		for (size_t i = 1; i < list.size(); i++)
+			if (list[i]->namePath() == list[i-1]->namePath())
+			{
+				SecondarySourceLocation ssl;
+				ssl.append("First occurrence here: ", list[i-1]->location());
+				m_errorReporter.typeError(
+					list[i]->location(),
+					ssl,
+					"Duplicate contract " +
+						joinHumanReadable(list[i]->namePath(), ".") +
+						" found in override list of " +
+						_function.name() +
+						"."
+				);
+			}
+	}
+
+	decltype(specifiedContracts) missingContracts;
+
+	// Iterate over the overrides
+	for (auto [begin, end] = _funcSet.equal_range(&_function); begin != end; begin++)
+	{
+		// Validate the override
+		if (!checkFunctionOverride(_function, **begin))
+			break;
+
+		auto const result = find(
+			specifiedContracts.cbegin(),
+			specifiedContracts.cend(),
+			(*begin)->annotation().contract
+		);
+
+		if (result == specifiedContracts.cend())
+			missingContracts.insert((*begin)->annotation().contract);
+		else
+			specifiedContracts.erase(result);
+	}
+
+	if (missingContracts.size() > 1)
+		overrideListError(
+			_function,
+			missingContracts,
+			"Function needs to specify overridden ",
+			""
+		);
+
+	if (!specifiedContracts.empty())
+		overrideListError(
+			_function,
+			specifiedContracts,
+			"Invalid ",
+			"specified in override list: "
+		);
+}
+
+ContractLevelChecker::FunctionSet const& ContractLevelChecker::getBaseFunctions(ContractDefinition const* _contract) const
```

Comment:
```suggestion
ContractLevelChecker::FunctionSet const& ContractLevelChecker::inheritedFunctions(ContractDefinition const* _contract) const
```

If you say `inherited`, it is clear that this includes functions from direct and from indirect bases.